documentation/how-to/prepare-for-production.md

```diff
@@ -0,0 +1,193 @@
+For observability, Superset can be integrated with [Canonical Observability Stack](https://charmhub.io/topics/canonical-observability-stack). This integration allows you to monitor metrics, logs, and events from Superset, enabling proactive health checks and performance analysis. Prometheus, Grafana and Loki are included as part of this observability suite.
+
+To deploy `cos-lite` and expose its endpoints as offers, follow these steps:
+
+```bash
+# Deploy the cos-lite bundle
+juju add-model cos
+juju deploy cos-lite --trust
+```
+
+```bash
+# Expose the cos integration endpoints
+juju offer prometheus:metrics-endpoint
+juju offer loki:logging
+juju offer grafana:grafana-dashboard
+
+# Relate superset to the cos-lite apps
+juju relate superset-k8s admin/cos.grafana
+juju relate superset-k8s admin/cos.loki
+juju relate superset-k8s admin/cos.prometheus
+```
+
+```bash
+# Access grafana with username "admin" and password
+juju run grafana/0 -m cos get-admin-password --wait 1m
+
+# Note the application IP address
+juju status
+```
+The Grafana UI can be found on the application IP address, port 3000. Pre-built dashboards are included under `Superset Metrics`.
+
+
+## Security
+
+### Terminate TLS at Ingress
+Superset can terminate TLS at the ingress by leveraging the [Nginx Ingress Integrator Charm](https://charmhub.io/nginx-ingress-integrator).
+
+You can deploy this with:
+
+```bash
+# Deploy the nginx ingress integrator charm
+juju deploy nginx-ingress-integrator --trust
+```
+
+#### Using K8s Secrets
+You can use a self-signed or production-grade TLS certificate stored in a Kubernetes secret. The secret is then associated with the ingress to encrypt traffic between clients and the Superset UI.
+
+For self-signed certificates you can do the below, if you have a production-grade certificate skip to the `create a k8s secret` step.
+
+```bash
+# Generate private key
+openssl genrsa -out server.key 2048
+
+# Generate a certificate signing request
+openssl req -new -key server.key -out server.csr -subj "/CN=<YOUR_HOST_NAME>"
+
+# Create self-signed certificate
+openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt -extfile <(printf "subjectAltName=DNS:<YOUR_HOST_NAME>")
+
+# Create a k8s secret
+kubectl create secret tls superset-tls --cert=server.crt --key=server.key
+```
+
+This secret name is provided to Charmed Superset and related to the Nginx Ingress Integrator:
+
+```bash
+# Configure the secret name
+juju config superset-k8s tls-secret-name=superset-tls
+
+# Configure the hostname
+juju config superset-k8s external-hostname=<YOUR_HOST_NAME>
+
+# Relate to the Integrator Charm
+juju relate superset-k8s nginx-ingress-integrator
+```
+
+You can now validate your ingress has been created with the tls certificates provided:
+```bash
+# Note the ingress with name like relation_<num>-<hostname>-ingress
+kubectl get ingress
+
+kubectl describe <YOUR_INGRESS_NAME>
+```
+You should see this using the K8s secret provided.
```

Comment:
See what? is an output/screenshot missing here?

Reply:
Updated